src/libstore/derivations.hh

```diff
@@ -40,18 +40,36 @@ typedef std::map<Path, StringSet> DerivationInputs;
 
 typedef std::map<string, string> StringPairs;
 
+
+/* A Derivation without its closure (sub-derivations).
+   This is used to prevent having to send the closure of a drv file
+   (which could be very large), e.g. for remote build operations. */
 struct BasicDerivation
 {
-    DerivationOutputs outputs; /* keyed on symbolic IDs */
-    PathSet inputSrcs; /* inputs that are sources */
+    /* Keyed on symbolic IDs, like `"out"` or `"doc"`. */
+    DerivationOutputs outputs;
+
+    /* Inputs that are sources. <- TODO what’s a source? */
+    PathSet inputSrcs;
```

Comment:
By process of elimination, if I have `cp ${./foobar} $out` then `./foobar` would be moved to the Nix store, and the Nix store path would be in the srcs.

Reply:
Yes, this is correct.